tests/python/relax/test_codegen_cudnn.py

```diff
@@ -0,0 +1,255 @@
+    data_shape, weight_shape, dtype, with_bias, activation
+):
+    low, high = -1, 1
+    data = np.random.randint(low, high, size=data_shape).astype(dtype)
+    weight = np.random.randint(low, high, size=weight_shape).astype(dtype)
+    bias = np.random.randint(low, high, size=(1, 1, 1, weight_shape[0])).astype(dtype)
+    activation = _activation_table[activation]
+    if with_bias:
+        args = (data, weight, bias)
+    else:
+        args = (data, weight)
+    mod = get_relax_conv2d_module(
+        data_shape,
+        weight_shape,
+        dtype,
+        with_bias=with_bias,
+        activation=activation,
+    )
+    mod = partition_for_cudnn(mod)
+    assert (
+        mod["main"].body.blocks[0].bindings[0].value.op.name_hint == "fused_relax_nn_conv2d_cudnn"
+    )
+
+
+@pytest.mark.parametrize(
+    "data_shape, weight_shape, dtype, with_bias, activation",
+    [
+        # Regular
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float32", False, "none"),
+        # Bias
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float32", True, "none"),
+        # Bias+ReLU
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float32", True, "relu"),
+        # Bias+ReLU+half
+        ((16, 32, 32, 16), (32, 3, 3, 16), "float16", True, "relu"),
+    ],
+)
+def test_conv2d_offload(data_shape, weight_shape, dtype, with_bias, activation):
+    input = np.random.randn(*data_shape).astype(dtype)
+    weight = np.random.randn(*weight_shape).astype(dtype)
+
+    if with_bias:
+        oc = weight_shape[0]
+        # bias = np.random.randn(oc).astype(dtype)
+        # bias = bias.reshape((1, 1, 1, weight_shape[0]))
+        bias = np.ones((1, 1, 1, weight_shape[0])).astype(dtype)
+        args = (input, weight, bias)
+    else:
+        bias = None
+        args = (input, weight)
+
+    activation = _activation_table[activation]
+
+    mod = get_relax_conv2d_module(
+        data_shape,
+        weight_shape,
+        dtype,
+        with_bias=with_bias,
+        activation=activation,
+    )
+
+    out = get_result_with_relax_cudnn_offload(mod, args)
+    ref = build_and_run(mod, args, "llvm", legalize=True)
+    if dtype == "float16":
+        tvm.testing.assert_allclose(out, ref, rtol=1e-1, atol=1e-1)
+    else:
+        tvm.testing.assert_allclose(out, ref, rtol=1e-2, atol=1e-2)
+
+
+@pytest.mark.parametrize(
+    "data_shape, weight_shape, dtype, with_bias, activation",
+    [
+        # Regular
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float32", False, "none"),
+        # Bias
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float32", True, "none"),
+        # Bias+ReLU
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float32", True, "relu"),
+        # Bias+ReLU+half
+        ((16, 16, 32, 32), (32, 16, 3, 3), "float16", True, "relu"),
+    ],
+)
+def test_conv2d_nchw_oihw_offload(data_shape, weight_shape, dtype, with_bias, activation):
+    input = np.random.randn(*data_shape).astype(dtype)
+    weight = np.random.randn(*weight_shape).astype(dtype)
```

Comment:
You can also try binding weights as a constant tensor: https://github.com/apache/tvm/blob/unity/tests/python/relax/test_codegen_tensorrt.py#L95

Reply:
I'll have a try, but what's the benefit?